src/PlatesEngineFactory.php

```diff
@@ -48,7 +49,15 @@ class PlatesEngineFactory
     public function __invoke(ContainerInterface $container): PlatesEngine
     {
         $config = $container->has('config') ? $container->get('config') : [];
-        $config = $config['plates'] ?? [];
+
+        $mezzioConfig = isset($config['templates']) && is_array($config['templates'])
+            ? $config['templates']
+            : [];
+        $platesConfig = isset($config['plates']) && is_array($config['plates'])
+            ? $config['plates']
+            : [];
+
+        $config = array_replace_recursive($mezzioConfig, $platesConfig);
```

Comment:
The entire part is confusing: `extensions` vs. `extension` vs. `file_extension` :confused:

Reply:
Its the same behaviour as here: https://github.com/mezzio/mezzio-twigrenderer/blob/57f301d01181f96f1379b8397ca285b856622efc/src/TwigRendererFactory.php#L59-L66

Which allows me to set e.g.:

```php
[
    'templates' => [
         'extension' => 'plates.php'
    ],
]
```

but when I'm using plates and twig currently this will be used for "both" template renderer. So twig allows currently to handle this via:

```php
[
    'templates' => [
         'extension' => 'tpl'
    ],
    'twig' => [
         'extension' => 'html.twig'
    ],
]
```

But as `twig` config provider sets the default of `templates.extension` to `html.twig` it can currently not be configured only for plates, which was very confusing.

This pull request implementes the same behaviour as already possible for Twig that you can define the extension only for plates template renderer and not general for all template renderers:

```php
[
    'plates' => [
         'extension' => 'plates.php'
    ],
    'twig' => [
         'extension' => 'html.twig'
    ],
]
```

The `templater renderer` abstraction was not keep in mind that there area inside project a project multiple template renderers of different engines. The current implementation is just a way to keep things backward compatible but allow to set the extension for `plates` templates.

My personally opinion is the whole `config` should not had not been abstracted and there should only be `twig` and `plates` config and the `templates` config part should be removed. But that is a very personally opinion as I'm implementing a lot wider abstraction in my [template abstraction project](https://github.com/schranz-templating/templating), and I having a lot different template engines in it I learned every template requires its own integration config and the config of them should not be abstracted.

Reply:
But yes a test should be added :)

Reply:
@Ocramius Test added.

@froschdesign naming did exist before, was not introduced by me :)

Reply:
@alexander-schranz 
I'm sorry, my comment was general and does not apply to your implementation.